letshelp-certbot/letshelp_certbot/magic_typing.py

```diff
@@ -0,0 +1,16 @@
+"""Shim class to not have to depend on typing module in prod."""
+import sys
+
+class TypingClass(object):
+    """Ignore import errors by getting anything"""
+    def __getattr__(self, name):
+        return None
+
+try:
+    # mypy doesn't respect modifying sys.modules
+    from typing import *  # pylint: disable=wildcard-import, unused-wildcard-import
+    # pylint: disable=unused-import
+    from typing import Collection, IO  # type: ignore
```

Comment:
what's up with this?

Reply:
This came up during the sprints.

[typing defines \_\_all\_\_](https://github.com/python/cpython/blob/3.6/Lib/typing.py#L24) which controls which names are imported when you do `from typing import *`. See https://docs.python.org/2/tutorial/modules.html#importing-from-a-package. We needed `Collection` and `IO` when fixing untyped defs for Certbot which aren't included in `__all__`.

I just copied the existing `acme.magic_typing` file here when creating this PR. I think there is some benefit in trying to keep this file somewhat consistent in the different places we have to copy it (there's a copy in `josepy` now too), but I can remove this line here if you prefer.

Reply:
if we add Collection and IO explicitly in `acme.magic_typing`, can we just have the one file then? or do we still need this? is there a problem with importing them explicitly there?

Reply:
> if we add Collection and IO explicitly in acme.magic_typing, can we just have the one file then?

I think we need at least one copy of `magic_typing` which is the copy here. While we're not currently using this package, I think it's beneficial to not have any dependencies to make it trivial for someone to download and use this.

If we really wanted, we can just make use of the `josepy.magic_typing` and remove `acme.magic_typing` so the file in `letshelp_certbot` is the only copy, but I believe you didn't want to do this before.

> is there a problem with importing them explicitly there?

Importing them explicitly hurts nothing. It just makes them available when `from typing import *` does not.